File: androidmvp/src/main/java/com/hsc/vince/androidmvp/base/MvpLazyFragment.java
package com.hsc.vince.androidmvp.base;

import android.content.Context;

/**
 * <p>作者：黄思程  2018/4/12 10:21
 * <p>邮箱：codinghuang@163.com
 * <p>作用：
 * <p>描述：带网络请求的懒加载Fragment
 */
public abstract class MvpLazyFragment<P extends BasePresenter> extends BaseLazyFragment {
    protected P mPresenter;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        mPresenter = createPresenter();
    }

    protected abstract P createPresenter();

    @Override
    public void onDetach() {
        super.onDetach();

        if (mPresenter != null) {
            mPresenter.detachView();
        }
    }
}

File: news/src/main/java/com/vince/news/base/MvpLazyFragment.kt
package com.vince.news.base

import android.os.Bundle
import android.view.View

/**
 * <p>作者：黄思程  2017/11/16 17:12
 * <p>邮箱：huangsicheng@gouuse.cn
 * <p>作用：
 * <p>描述：带网络请求的懒加载Fragment
 */
abstract class MvpLazyFragment<P : BasePresenter<*>> : BaseLazyFragment() {
    protected lateinit var mvpPresenter: P

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        mvpPresenter = createPresenter()

        super.onViewCreated(view, savedInstanceState)
    }

    override fun onDestroyView() {
        super.onDestroyView()

        mvpPresenter.detachView()
    }

    /**
     * @return 创建Presenter
     * */
    protected abstract fun createPresenter(): P
}